### PR TITLE
RELATED: ONE-4434 Fix caching to avoid duplicates

### PR DESCRIPTION
**Pull request by pbenes**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [x] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/3057
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/2777

# Related Jira tasks

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7166/ : SUCCESS in 5m 44s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3136/ : SUCCESS in 4s

**Comment by pbenes** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3747/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3137/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3120/ : SUCCESS in 13m 18s

**Comment by pbenes** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1568.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3138/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3121/ : SUCCESS in 18m 36s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7167/ : SUCCESS in 4m 16s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3139/ : SUCCESS in 5s

File: src/components/core/pivotTable/agGridDataSourceUtils.ts [hunk 1]
```diff
@@ -60,17 +60,20 @@ export const wrapGetPageWithCaching = (getPage: IGetPage): IGetPage => {
...
+
         if (firstCachedPageRequest === null) {
             firstCachedPageRequest = {
                 requestParams,
-                response: execution,
+                resultPromise,
             };
         }
 
+        const execution = await resultPromise;
```
Comment (patrikbraborec) [inline src/components/core/pivotTable/agGridDataSourceUtils.ts:76]: Should be here just `return await resultPromise`? :)

Comment (pbenes) [reply src/components/core/pivotTable/agGridDataSourceUtils.ts:76]: It breaks the no-return-await eslint rule (https://eslint.org/docs/rules/no-return-await)

Comment (patrikbraborec) [reply src/components/core/pivotTable/agGridDataSourceUtils.ts:76]: Oh, I see. Thanks! :)

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3749/ : SUCCESS in 24m 07s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7176/ : SUCCESS in 2m 53s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1746/ : SUCCESS in 14s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2564/ : SUCCESS in 19m 51s

**Pull request merged**